Framework/jmaqs-appium/src/test/java/com/magenic/jmaqs/appium/AppiumDriverFactoryTest.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2019 (C) Magenic, All rights Reserved
+ */
+
+package com.magenic.jmaqs.appium;
+
+import io.appium.java_client.AppiumDriver;
+import io.appium.java_client.remote.MobileCapabilityType;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.remote.DesiredCapabilities;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/**
+ * The type Appium driver factory test.
+ */
+public class AppiumDriverFactoryTest {
+
+  private static final String MESSAGE = "Test method not implemented yet.";
+
+  private static DesiredCapabilities sauceLabsConfig;
+
+  /**
+   * Sets up.
+   */
+  @BeforeClass
+  public void setUp() {
+    sauceLabsConfig = new DesiredCapabilities();
+    sauceLabsConfig.setCapability("username", "Partner_Magenic");
+    sauceLabsConfig.setCapability("accessKey", "0a4d7d84-f93b-43e6-9af1-1c89ac143355");
+    sauceLabsConfig.setCapability("deviceOrientation", "portrait");
+  }
+
+  /**
+   * Test get default mobile driver.
+   */
+  @Test
+  public void testGetDefaultMobileDriver() {
+    AppiumDriver<WebElement> defaultMobileDriver = AppiumDriverFactory.getDefaultMobileDriver();
+    Assert.assertNotNull(defaultMobileDriver, "Checking if default driver is null");
+  }
+
+  /**
+   * Test test get default mobile driver android.
+   */
+  @Test
+  public void testTestGetDefaultMobileDriverAndroid() {
+    AppiumDriver<WebElement> defaultMobileDriver = AppiumDriverFactory
+        .getDefaultMobileDriver(PlatformType.ANDROID);
+    Assert.assertNotNull(defaultMobileDriver, "Checking if default driver is null");
+  }
+
+  /**
+   * Test get default mobile options.
+   */
+  @Test
+  public void testGetDefaultMobileOptions() {
+    final DesiredCapabilities defaultMobileOptions = AppiumDriverFactory.getDefaultMobileOptions();
+    Consumer<String> assertionConsumer = (String s) -> {
+      Assert.assertNotNull(defaultMobileOptions.is(s),
+          String.format("Checking if capability key %s is not null", s));
+    };
+    defaultMobileOptions.getCapabilityNames().forEach(assertionConsumer);
+  }
+
+  /**
+   * Test test get default mobile options.
+   */
+  @Test
+  public void testTestGetDefaultMobileOptions() {
+    final Map<String, Object> capabilitiesAsObjects = AppiumConfig.getCapabilitiesAsObjects();
+    DesiredCapabilities capabilities = AppiumDriverFactory
+        .getDefaultMobileOptions(capabilitiesAsObjects);
+    Consumer<String> assertionConsumer = (String s) -> {
```

Comment:
This in test Consumer is fine in a Unit Test, but we should add a comment if we do this inline within unit tests.